### PR TITLE
IGNITE-12285 Removes boilerplate code in test PluginProvider implementations.

### PR DESCRIPTION
**Pull request by ololo3000**

### CODE REVIEW

File: modules/extdata/platform/pom.xml [hunk 1]
```diff
@@ -42,5 +42,13 @@
             <version>${project.version}</version>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.ignite</groupId>
+            <artifactId>ignite-core</artifactId>
```
Comment (NSAmelchev) [inline modules/extdata/platform/pom.xml]: Why dependency duplicated?

Comment (ololo3000) [reply modules/extdata/platform/pom.xml]: There is no duplication. The first dependency for ignite-core production classes, the second for corresponding test classes.

File: modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import java.io.Serializable;
+import java.util.UUID;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.cluster.ClusterNode;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * No-op test implementation of {@link PluginProvider} which allows to avoid redundant boilerplate code.
+ */
+public abstract class AbstractTestPluginProvider<C extends PluginConfiguration> implements PluginProvider<C> {
```
Comment (dgarus) [inline modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java]: I think we can make PluginProvider<PluginConfiguration> and don't use a generic parameter in inherited classes.

File: modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    /** {@inheritDoc} */
+    @Override public String version() {
+        return null;
```
[the remark this answers is not in the document]
Comment (ololo3000) [reply modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java]: Done.

File: modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java (hunk 2 above)
Comment (ololo3000) [reply modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java]: Done.

File: modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        // No-op.
+    }
+
+    /** {@inheritDoc} */
+    @Override public void stop(boolean cancel) throws IgniteCheckedException {
+        // No-op.
+    }
+
+    /** {@inheritDoc} */
+    @Override public void onIgniteStart() throws IgniteCheckedException {
```
[the remark this answers is not in the document]
Comment (ololo3000) [reply modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java]: Done.

File: modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override public String copyright() {
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void initExtensions(PluginContext ctx, ExtensionRegistry registry) throws IgniteCheckedException {
```
[the remark this answers is not in the document]
Comment (ololo3000) [reply modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java]: Done.

File: modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import java.io.Serializable;
+import java.util.UUID;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.cluster.ClusterNode;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * No-op test implementation of {@link PluginProvider} which allows to avoid redundant boilerplate code.
+ */
+public abstract class AbstractTestPluginProvider implements PluginProvider<PluginConfiguration> {
```
Comment (dgarus) [inline modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java]: We can make the `plugin` method implementation here and delete a few codes in inherited classes.
`@Override public <T extends IgnitePlugin> T plugin() {
        return (T)new IgnitePlugin() {
            // No-op.
        };
    }`

Comment (ololo3000) [reply modules/core/src/test/java/org/apache/ignite/plugin/AbstractTestPluginProvider.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/plugin/PluginProvider.java [hunk 7]
```diff
@@ -28,11 +28,13 @@
 /**
  * Pluggable Ignite component.
  * <p>
- * Ignite plugins are loaded using JDK {@link ServiceLoader}.
+ * Ignite plugins can be loaded using JDK {@link ServiceLoader} or set up explicitly via
```
Comment (andrey-kuznetsov) [inline modules/core/src/main/java/org/apache/ignite/plugin/PluginProvider.java]: Please mention the class that should be exposed as a service (i.e. PluginProvider.class itself).

**Pull request merged**